torch/csrc/jit/passes/quantization.cpp

```diff
@@ -1539,7 +1539,7 @@ void insertDeQuantForAllUse(
     // in the same block so that quant fusion can happen
     WithInsertPoint ins(user);
     Node* dequant = insertDeQuant(graph, quantized_val, original_val, i);
-    user->replaceInputWith(original_val, dequant->output());
+    user->replaceInput(uses[i].offset, dequant->output());
```

Comment:
Dont have the whole context here, but usually this means we missed a test that should have identified this issue. Is there a simple test that can be added?

Reply:
Yeah I think it's better to add a test.
this is pretty rare, it only happens when we do something like `torch.mul(x, x)`